### ISSUE TITLE
Support conditional checks

### ISSUE THREAD

**Issue by fhunleth** [nerves-project/nerves_system_linter#9]
For example, if BR2_TOOLCHAIN_USES_GLIBC=y, then BR2_GENERATE_LOCALE should be false. Non-glibc systems are currently failing since they won't have BR2_GENERATE_LOCALE.

**Comment by ConnorRigby** [comment]
I'm gonna take a look at this this weekend :tm: I'm not 100% sure how i will implement it, but i have some ideas.

**Comment by ConnorRigby** [comment]
```elixir
defmodule Nerves.System.Linter.Rule.GenerateLocale do
  use Nerves.System.Linter.Rule

  ensure_value("BR2_GENERATE_LOCALE", "en_US.UTF-8", [
    message: "BR2_GENERATE_LOCALE should be en_US.UTF-8 if BR2_TOOLCHAIN_USES_GLIBC is disabeld.",
    if: [{"BR2_TOOLCHAIN_USES_GLIBC", false}]
  ])

  ensure_value("BR2_GENERATE_LOCALE", false, [
    message: "BR2_GENERATE_LOCALE should be disabled if BR2_TOOLCHAIN_USES_GLIBC is enabled",
    if: [{"BR2_TOOLCHAIN_USES_GLIBC", true}]
  ])

  evaluate()
end

```

**Comment by ConnorRigby** [comment]
Fixed with #10
